prime-router/src/main/kotlin/ActionMessages.kt

```diff
@@ -140,4 +140,26 @@ class InvalidReportMessage(override val message: String) : GenericActionLogDetai
  * A [message] for invalid translation operation.
  */
 class InvalidTranslationMessage(override val message: String) :
-    GenericActionLogDetail(message, ActionLogScope.translation)
\ No newline at end of file
+    GenericActionLogDetail(message, ActionLogScope.translation)
+
+/**
+ * A [message] for entire duplicate report submission
+ */
+class DuplicateFileMessage(val payloadName: String?, val allItemsDupe: Boolean) : ActionLogDetail {
+    override val scope = ActionLogScope.report
+    override val message: String get() {
+        var msg = if (allItemsDupe) "All items in this file are duplicates." else "Duplicate file detected."
+        if (!payloadName.isNullOrEmpty()) {
+            msg += " File: $payloadName"
+        }
+        return msg
+    }
+}
+
+/**
+ * A [message] for a duplicate item within a report
+ */
+class DuplicateItemMessage(val rowNum: Int) : ActionLogDetail {
+    override val scope = ActionLogScope.item
+    override val message = "Item $rowNum is a duplicate."
```

Comment:
You need to change this, so the indices array is done correctly in the response.  Make this like on line 127 and remove the rowNum from the message.  You already handle the item log correctly where you use this, so no changes there.

Reply:
This is directly counter to the requirement from Jim to include rowNum in the response message. It is returned both in the message and using the 'indices' property.